erts/emulator/beam/erl_map.c

```diff
@@ -202,25 +203,33 @@ BIF_RETTYPE maps_find_2(BIF_ALIST_2) {
     BIF_ERROR(BIF_P, BADMAP);
 }
 
-/* maps:get/2
+/* maps:get/2 and erlang:map_get/2
  * return value if key *matches* a key in the map
  * exception badkey if none matches
  */
 
-BIF_RETTYPE maps_get_2(BIF_ALIST_2) {
-    if (is_map(BIF_ARG_2)) {
+BIF_RETTYPE ERTS_INLINE common_maps_get(Process *p, Eterm key, Eterm map) {
```

Comment:
This function should be static, or put the entire implementation into maps_get_2 and make map_get_2 call it.

Reply:
I tried to make one call the other but I couldn't figure out all the arguments to satisfy the BIF_ALIST_2 declaration.

Reply:
Use the BIF_CALL_ARGS macro. i.e. something like:

```
BIF_RETTYPE map_get_2(BIF_ALIST_2) {
  BIF_RET(maps_get_2(BIF_CALL_ARGS));
}
```